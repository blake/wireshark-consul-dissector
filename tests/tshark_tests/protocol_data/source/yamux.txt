### Yamux test data

# RPC Type: MultiplexV2 (0x04)
# Direction: Inbound (I)
I
# Packet bytes
0000   04

#### Data ####

# Message type: Data (0x00)
# Direction: Inbound (I)
I
# Packet bytes
0000   00 00 00 00 00 00 00 01 00 00 00 26

# Message type: Data (0x00)
# Direction: Outbound (0)
O
# Packet bytes
0000   00 00 00 00 00 00 00 01 00 00 00 26

#### Window Update ####
# Message type: Window Update (0x01)
# Flags: SYN (0x0001)
# Direction: Outbound (O)
O
0000   00 01 00 01 00 00 00 01 00 00 00 00

# Message type: Window Update (0x01)
# Flags: ACK (0x0002)
# Direction: Inbound (I)
I
0000   00 01 00 02 00 00 00 01 00 00 00 00

# Message type: Window Update (0x01)
# Flags: SYN (0x0001)
# Direction: Inbound (I)
I
0000   00 01 00 01 00 00 00 01 00 00 00 00

# Message type: Window Update (0x01)
# Flags: ACK (0x0002)
# Direction: Outbound (O)
O
0000   00 01 00 02 00 00 00 01 00 00 00 00

# Message type: Window Update (0x01)
# Flags: FIN (0x0004)
# Direction: Outbound (O)
O
0000   00 01 00 04 00 00 00 03 00 00 00 00

# Message type: Window Update (0x01)
# Flags: FIN (0x0004)
# Direction: Inbound (I)
I
0000   00 01 00 04 00 00 00 03 00 00 00 00

#### Ping ####
# Message type: Ping (0x02)
# Flags: SYN (0x0001)
# Direction: Outbound (O)
O
0000   00 02 00 01 00 00 00 00 00 00 00 04

# Message type: Ping (0x02)
# Flags: ACK (0x0002)
# Direction: Inbound (O)
I
0000   00 02 00 02 00 00 00 00 00 00 00 04

# Message type: Ping (0x02)
# Flags: SYN (0x0001)
# Direction: Inbound (I)
I
0000   00 02 00 01 00 00 00 00 00 00 00 04

# Message type: Ping (0x02)
# Flags: ACK (0x0002)
# Direction: Outbound (O)
O
0000   00 02 00 02 00 00 00 00 00 00 00 04

#### Go Away ####
# Message type: GoAway (0x03)
# Direction: Outbound (O)
# O
# <bytes>

# Message type: GoAway (0x03)
# Direction: Inbound (I)
# I
# <bytes>
