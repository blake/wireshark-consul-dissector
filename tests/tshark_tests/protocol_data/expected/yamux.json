[
  {
    "_type": "doc",
    "_score": null,
    "_source": {
      "layers": {
        "frame": {
          "filtered": "frame"
        },
        "eth": {
          "filtered": "eth"
        },
        "ip": {
          "filtered": "ip"
        },
        "tcp": {
          "filtered": "tcp"
        },
        "_ws.lua.fake": "",
        "consul": {
          "_ws.lua.fake": ""
        }
      }
    }
  },
  {
    "_type": "doc",
    "_score": null,
    "_source": {
      "layers": {
        "frame": {
          "filtered": "frame"
        },
        "eth": {
          "filtered": "eth"
        },
        "ip": {
          "filtered": "ip"
        },
        "tcp": {
          "filtered": "tcp"
        },
        "_ws.lua.fake": "",
        "yamux": {
          "yamux.version": "0",
          "yamux.type": "0x00",
          "yamux.flags": "0x0000",
          "yamux.stream_id": "1",
          "yamux.length": "38",
          "yamux.payload_length": "38"
        }
      }
    }
  },
  {
    "_type": "doc",
    "_score": null,
    "_source": {
      "layers": {
        "frame": {
          "filtered": "frame"
        },
        "eth": {
          "filtered": "eth"
        },
        "ip": {
          "filtered": "ip"
        },
        "tcp": {
          "filtered": "tcp"
        },
        "_ws.lua.fake": "",
        "yamux": {
          "yamux.version": "0",
          "yamux.type": "0x00",
          "yamux.flags": "0x0000",
          "yamux.stream_id": "1",
          "yamux.length": "38",
          "yamux.payload_length": "38",
          "yamux.previous_frame": "2",
          "_ws.expert": {
            "yamux.expert.debug.corrupted_frame": "",
            "_ws.expert.message": "Previous frame is corrupted: no window size is available",
            "_ws.expert.severity": "6291456",
            "_ws.expert.group": "134217728"
          },
          "yamux.window_size.server.before": "262144",
          "yamux.window_size.client.before": "262144",
          "_ws.expert": {
            "yamux.expert.data_before_start": "",
            "_ws.expert.message": "Stream packets were detected without SYN flag",
            "_ws.expert.severity": "8388608",
            "_ws.expert.group": "33554432"
          },
          "yamux.window_size.server.after": "262144",
          "yamux.window_size.client.after": "262144"
        }
      }
    }
  },
  {
    "_type": "doc",
    "_score": null,
    "_source": {
      "layers": {
        "frame": {
          "filtered": "frame"
        },
        "eth": {
          "filtered": "eth"
        },
        "ip": {
          "filtered": "ip"
        },
        "tcp": {
          "filtered": "tcp"
        },
        "_ws.lua.fake": "",
        "yamux": {
          "yamux.version": "0",
          "yamux.type": "0x01",
          "yamux.flags": "0x0001",
          "yamux.stream_id": "1",
          "yamux.length": "0",
          "yamux.recv_window_delta": "0",
          "yamux.previous_frame": "3",
          "yamux.window_size.server.before": "262144",
          "yamux.window_size.client.before": "262144",
          "_ws.expert": {
            "yamux.expert.packet_type.client": "",
            "_ws.expert.message": "Current packet is generated by client",
            "_ws.expert.severity": "1048576",
            "_ws.expert.group": "134217728"
          },
          "yamux.window_size.server.after": "262144",
          "yamux.window_size.client.after": "262144"
        }
      }
    }
  },
  {
    "_type": "doc",
    "_score": null,
    "_source": {
      "layers": {
        "frame": {
          "filtered": "frame"
        },
        "eth": {
          "filtered": "eth"
        },
        "ip": {
          "filtered": "ip"
        },
        "tcp": {
          "filtered": "tcp"
        },
        "_ws.lua.fake": "",
        "yamux": {
          "yamux.version": "0",
          "yamux.type": "0x01",
          "yamux.flags": "0x0002",
          "yamux.stream_id": "1",
          "yamux.length": "0",
          "yamux.recv_window_delta": "0",
          "yamux.previous_frame": "4",
          "yamux.window_size.server.before": "262144",
          "yamux.window_size.client.before": "262144",
          "_ws.expert": {
            "yamux.expert.packet_type.client": "",
            "_ws.expert.message": "Current packet is generated by client",
            "_ws.expert.severity": "1048576",
            "_ws.expert.group": "134217728"
          },
          "yamux.window_size.server.after": "262144",
          "yamux.window_size.client.after": "262144"
        }
      }
    }
  },
  {
    "_type": "doc",
    "_score": null,
    "_source": {
      "layers": {
        "frame": {
          "filtered": "frame"
        },
        "eth": {
          "filtered": "eth"
        },
        "ip": {
          "filtered": "ip"
        },
        "tcp": {
          "filtered": "tcp"
        },
        "_ws.lua.fake": "",
        "yamux": {
          "yamux.version": "0",
          "yamux.type": "0x01",
          "yamux.flags": "0x0001",
          "yamux.stream_id": "1",
          "yamux.length": "0",
          "yamux.recv_window_delta": "0",
          "yamux.previous_frame": "5",
          "yamux.window_size.server.before": "262144",
          "yamux.window_size.client.before": "262144",
          "_ws.expert": {
            "yamux.expert.packet_type.client": "",
            "_ws.expert.message": "Current packet is generated by client",
            "_ws.expert.severity": "1048576",
            "_ws.expert.group": "134217728"
          },
          "yamux.window_size.server.after": "262144",
          "yamux.window_size.client.after": "262144"
        }
      }
    }
  },
  {
    "_type": "doc",
    "_score": null,
    "_source": {
      "layers": {
        "frame": {
          "filtered": "frame"
        },
        "eth": {
          "filtered": "eth"
        },
        "ip": {
          "filtered": "ip"
        },
        "tcp": {
          "filtered": "tcp"
        },
        "_ws.lua.fake": "",
        "yamux": {
          "yamux.version": "0",
          "yamux.type": "0x01",
          "yamux.flags": "0x0002",
          "yamux.stream_id": "1",
          "yamux.length": "0",
          "yamux.recv_window_delta": "0",
          "yamux.previous_frame": "6",
          "yamux.window_size.server.before": "262144",
          "yamux.window_size.client.before": "262144",
          "_ws.expert": {
            "yamux.expert.packet_type.client": "",
            "_ws.expert.message": "Current packet is generated by client",
            "_ws.expert.severity": "1048576",
            "_ws.expert.group": "134217728"
          },
          "yamux.window_size.server.after": "262144",
          "yamux.window_size.client.after": "262144"
        }
      }
    }
  },
  {
    "_type": "doc",
    "_score": null,
    "_source": {
      "layers": {
        "frame": {
          "filtered": "frame"
        },
        "eth": {
          "filtered": "eth"
        },
        "ip": {
          "filtered": "ip"
        },
        "tcp": {
          "filtered": "tcp"
        },
        "_ws.lua.fake": "",
        "yamux": {
          "yamux.version": "0",
          "yamux.type": "0x01",
          "yamux.flags": "0x0004",
          "yamux.stream_id": "3",
          "yamux.length": "0",
          "yamux.recv_window_delta": "0"
        }
      }
    }
  },
  {
    "_type": "doc",
    "_score": null,
    "_source": {
      "layers": {
        "frame": {
          "filtered": "frame"
        },
        "eth": {
          "filtered": "eth"
        },
        "ip": {
          "filtered": "ip"
        },
        "tcp": {
          "filtered": "tcp"
        },
        "_ws.lua.fake": "",
        "yamux": {
          "yamux.version": "0",
          "yamux.type": "0x01",
          "yamux.flags": "0x0004",
          "yamux.stream_id": "3",
          "yamux.length": "0",
          "yamux.recv_window_delta": "0",
          "yamux.previous_frame": "8",
          "_ws.expert": {
            "yamux.expert.debug.corrupted_frame": "",
            "_ws.expert.message": "Previous frame is corrupted: no window size is available",
            "_ws.expert.severity": "6291456",
            "_ws.expert.group": "134217728"
          },
          "yamux.window_size.server.before": "262144",
          "yamux.window_size.client.before": "262144",
          "_ws.expert": {
            "yamux.expert.data_before_start": "",
            "_ws.expert.message": "Stream packets were detected without SYN flag",
            "_ws.expert.severity": "8388608",
            "_ws.expert.group": "33554432"
          },
          "yamux.window_size.server.after": "262144",
          "yamux.window_size.client.after": "262144"
        }
      }
    }
  },
  {
    "_type": "doc",
    "_score": null,
    "_source": {
      "layers": {
        "frame": {
          "filtered": "frame"
        },
        "eth": {
          "filtered": "eth"
        },
        "ip": {
          "filtered": "ip"
        },
        "tcp": {
          "filtered": "tcp"
        },
        "_ws.lua.fake": "",
        "yamux": {
          "yamux.version": "0",
          "yamux.type": "0x02",
          "yamux.flags": "0x0001",
          "yamux.stream_id": "0",
          "yamux.length": "4",
          "yamux.ping_payload": "0x00000004",
          "_ws.expert": {
            "yamux.expert.open_default_stream": "",
            "_ws.expert.message": "Default opened stream cannot be opened",
            "_ws.expert.severity": "6291456",
            "_ws.expert.group": "117440512"
          },
          "_ws.malformed": "Malformed Packet"
        }
      }
    }
  },
  {
    "_type": "doc",
    "_score": null,
    "_source": {
      "layers": {
        "frame": {
          "filtered": "frame"
        },
        "eth": {
          "filtered": "eth"
        },
        "ip": {
          "filtered": "ip"
        },
        "tcp": {
          "filtered": "tcp"
        },
        "_ws.lua.fake": "",
        "yamux": {
          "yamux.version": "0",
          "yamux.type": "0x02",
          "yamux.flags": "0x0002",
          "yamux.stream_id": "0",
          "yamux.length": "4",
          "yamux.ping_payload": "0x00000004",
          "yamux.previous_frame": "10"
        }
      }
    }
  },
  {
    "_type": "doc",
    "_score": null,
    "_source": {
      "layers": {
        "frame": {
          "filtered": "frame"
        },
        "eth": {
          "filtered": "eth"
        },
        "ip": {
          "filtered": "ip"
        },
        "tcp": {
          "filtered": "tcp"
        },
        "_ws.lua.fake": "",
        "yamux": {
          "yamux.version": "0",
          "yamux.type": "0x02",
          "yamux.flags": "0x0001",
          "yamux.stream_id": "0",
          "yamux.length": "4",
          "yamux.ping_payload": "0x00000004",
          "_ws.expert": {
            "yamux.expert.open_default_stream": "",
            "_ws.expert.message": "Default opened stream cannot be opened",
            "_ws.expert.severity": "6291456",
            "_ws.expert.group": "117440512"
          },
          "_ws.malformed": "Malformed Packet",
          "yamux.previous_frame": "11"
        }
      }
    }
  },
  {
    "_type": "doc",
    "_score": null,
    "_source": {
      "layers": {
        "frame": {
          "filtered": "frame"
        },
        "eth": {
          "filtered": "eth"
        },
        "ip": {
          "filtered": "ip"
        },
        "tcp": {
          "filtered": "tcp"
        },
        "_ws.lua.fake": "",
        "yamux": {
          "yamux.version": "0",
          "yamux.type": "0x02",
          "yamux.flags": "0x0002",
          "yamux.stream_id": "0",
          "yamux.length": "4",
          "yamux.ping_payload": "0x00000004",
          "yamux.previous_frame": "12"
        }
      }
    }
  }
]
